test-project/test_contract/lib.rs

```diff
@@ -28,6 +28,8 @@ mod test_contract {
         #[ink(topic)]
         a: u32,
         b: Struct2,
+        c: [u64; 4],
+        d: (Struct1, Struct2),
```

Comment:
IIUC, this isn't special to the event itself - if we had `[u64; 4]` defined on the contract's `struct` we would have hit the same issue, right?

Reply:
Not really, because there we don't need the `super::` at any point. We have all the types either in the namespace of the contract, the global namespace or somewhere in `ink_primitives`. The whole problem arises because events need their own namespace so that we can safely use the name `Event` for the generated enum.